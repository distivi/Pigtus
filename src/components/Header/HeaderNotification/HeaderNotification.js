import React from 'react';
import PropsTypes from 'prop-types';

export default class HeaderNotification extends React.Component {
  getNotificationCount = () => {
    return this.props.count ?
      <div className="notification-counter">{this.props.count}</div> :
      '';
  }

  openNotification = () => {
    const callback = this.props.openNotification;

    if (callback) {
      callback();
    } else {
      console.warn('No notification callback provided.');
    }
  }

  render() {
    const count = this.getNotificationCount();
    return (
        <div className="header-notification"
          onClick={this.openNotification}>
          {count}
        </div>
    );
  }
}

HeaderNotification.propTypes = {
  count: PropsTypes.number,
  openNotification: PropsTypes.func
};
