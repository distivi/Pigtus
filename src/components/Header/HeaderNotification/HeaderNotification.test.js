import React from 'react';
import ReactDOM from 'react-dom';
import HeaderNotification from './HeaderNotification';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import sinon from 'sinon';

describe('<HeaderNotification/>', () => {
  it('renders HeaderNotification without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<HeaderNotification />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('HeaderNotification is consistent with mock', () => {
    const component = renderer.create(<HeaderNotification/>);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('HeaderNotification displays 2 new notifications', () => {
    const notification = shallow(<HeaderNotification count={2}/>);

    expect(notification.text()).toEqual('2');
  });

  it('HeaderNotification displays nothing in case count=0', () => {
    const notification = shallow(<HeaderNotification count={0}/>);

    expect(notification.text()).toEqual('');
  });


  it('HeaderNotification displays nothing in case no count property passed', () => {
    const notification = shallow(<HeaderNotification count={0}/>);

    expect(notification.text()).toEqual('');
  });

  it('HeaderNotification openNotification callback called', () => {
    const openNotificationCallback = sinon.spy();

    const notification = shallow(<HeaderNotification count={0} openNotification={openNotificationCallback}/>);
    expect(openNotificationCallback.calledOnce).toBeFalsy();
    notification.simulate('click');
    expect(openNotificationCallback.calledOnce).toBeTruthy();
  });
});
