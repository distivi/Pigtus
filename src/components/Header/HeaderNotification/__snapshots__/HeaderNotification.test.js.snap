// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderNotification/> HeaderNotification is consistent with mock 1`] = `
<div
  className="header-notification"
  onClick={[Function]}
>
  
</div>
`;
